core/src/Http/HttpRequest.cs

```diff
@@ -74,16 +80,39 @@ private static HttpRequestMessage CreateRequestMessage(Uri endpoint, Dictionary<
             return requestMessage;
         }
 
+        private static async Task<HttpContent> CloneContent(HttpContent content)
+        {
+            var temp = new System.IO.MemoryStream();
+            await content.CopyToAsync(temp).ConfigureAwait(false);
```

Comment:
>content [](start = 18, length = 7)

doesn't content have a deep copy? is this needed?

Reply:
`HttpContent` has a `MemberWiseClone()` doing the shallow copy, but [does not seem to have a deep copy](https://msdn.microsoft.com/en-us/library/system.net.http.httpcontent(v=vs.110).aspx#Anchor_4). My current implementation was inspired by [this StackOverflow answer](https://stackoverflow.com/a/34049029).

Reply:
I answered this days ago in a github comment, which was not showed as a reply to your original comment in CodeFlow. So I'm copying that reply here:
`HttpContent` has a `MemberWiseClone()` doing the shallow copy, but [does not seem to have a deep copy](https://msdn.microsoft.com/en-us/library/system.net.http.httpcontent(v=vs.110).aspx#Anchor_4). My current implementation was inspired by [this StackOverflow answer](https://stackoverflow.com/a/34049029).

---
In reply to: [201542733](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118#discussion_r201542733) [](ancestors = 201542733)